engine/execution/ingestion/engine.go

```diff
@@ -467,13 +454,14 @@ func (e *Engine) enqueueBlockAndCheckExecutable(
 	}
 
 	firstUnexecutedHeight := queue.Head.Item.Height()
-	if checkStateSync {
-		// whenever the queue grows, we need to check whether the state sync should be
-		// triggered.
-		e.unit.Launch(func() {
-			e.checkStateSyncStart(firstUnexecutedHeight)
-		})
-	}
+	// disable state syncing for now
+	// if checkStateSync {
+	// 	// whenever the queue grows, we need to check whether the state sync should be
+	// 	// triggered.
+	// 	e.unit.Launch(func() {
+	// 		e.checkStateSyncStart(firstUnexecutedHeight)
+	// 	})
+	// }
```

Comment:
shouldn't we just add a flag and set default to false for now, we might need to use this code for debugging

Reply:
I'm afraid there would still be a possibility this flag was set as true, and increasing the scope for debugging fires.